contract/contracts/StakingPool.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity ^0.6.0;
+
+import "./interfaces.sol";
+import "./FanToken.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Owner must be TokenFactory
+contract StakingPool is PoolInterface, Ownable {
+    using SafeMath for uint256;
+
+    // TODO define events
+
+    mapping(address  => uint256) public tokensTotalSupply;
+    mapping(address => mapping(address => uint256)) private creatorsTokenBalances;
+    mapping(address => bool) public registeredTokens;
+
+    function balanceOf(address account, address token) public view returns(uint256){
+        return creatorsTokenBalances[account][token];
+    }
+
+    function addTokenToStakingList(address token) public onlyOwner {
+        registeredTokens[token] = true;
```

Comment:
I think we've made the specification to allow staking to be stopped, so it should be changeable to false.

Reply:
And I think the creator should be the one who can change that state.

Reply:
You are right. I'm gonna fix it in other PR.